csrc/layer_norm_cuda_kernel.cu

```diff
@@ -908,9 +908,13 @@ void HostApplyRMSNorm(
 {
     auto stream = at::cuda::getCurrentCUDAStream().stream();
     const int warp_size = at::cuda::getCurrentDeviceProperties()->warpSize;
-    const dim3 threads(32,4,1);
     const uint64_t maxGridY = at::cuda::getCurrentDeviceProperties()->maxGridSize[1];
     const dim3 blocks(1, std::min((uint64_t)n1, maxGridY), 1);
+    dim3 threads(warp_size,4,1);
+    #ifdef __HIP_PLATFORM_HCC__
+    // Optimization for ROCm MI100
+    threads.y = 2;
```

Comment:
Is this only applicable for MI100 as per comment?

Reply:
It is for a perf ticket for MI100 in onnxruntime. It will need to be tested for MI200 or other hardware.